### PR TITLE
Allow the recursive directory iterator to iterate symbolic links

### PR DESCRIPTION
**Pull request by sherakama**
Directory iterator is not iterating through symbolic links as it did in behat 1.x and most of behat 2.x.
The new RecursiveDirectoryIterator needs to be explicitly told to follow symlinks or it will not.

### CODE REVIEW

File: src/Behat/Behat/Gherkin/Specification/Locator/FilesystemFeatureLocator.php [hunk 1]
```diff
@@ -131,13 +131,19 @@ private function findFeatureFiles($path)
             return array($absolutePath);
         }
 
-        $iterator = new RegexIterator(
-            new RecursiveIteratorIterator(
-                new RecursiveDirectoryIterator($absolutePath)
-            ), '/^.+\.feature$/i',
+        $RDI = new RecursiveDirectoryIterator(
```
Comment (sroze) [inline src/Behat/Behat/Gherkin/Specification/Locator/FilesystemFeatureLocator.php]: I'm not fond of these short variable names.

File: src/Behat/Behat/Gherkin/Specification/Locator/FilesystemFeatureLocator.php [hunk 2]
```diff
@@ -131,13 +131,19 @@ private function findFeatureFiles($path)
...
+            $absolutePath,
+            RecursiveDirectoryIterator::FOLLOW_SYMLINKS |
+            RecursiveDirectoryIterator::SKIP_DOTS
+        );
+        $RII = new recursiveIteratorIterator($RDI);
```
Comment (sroze) [inline src/Behat/Behat/Gherkin/Specification/Locator/FilesystemFeatureLocator.php]: You could simply construct them without any other variable like it was before btw.

**Comment by sherakama** [comment]
@sroze,

Thanks for reviewing this. I have restored the original formatting. It was easier to debug when I had them separated out in to separate variables. I agree, we don't need them.

File: src/Behat/Behat/Gherkin/Specification/Locator/FilesystemFeatureLocator.php [hunk 3]
```diff
@@ -132,11 +132,17 @@ private function findFeatureFiles($path)
         }
 
         $iterator = new RegexIterator(
-            new RecursiveIteratorIterator(
-                new RecursiveDirectoryIterator($absolutePath)
-            ), '/^.+\.feature$/i',
+            new recursiveIteratorIterator(
```
Comment (sroze) [inline src/Behat/Behat/Gherkin/Specification/Locator/FilesystemFeatureLocator.php]: You should use an uppercase `R` else that will fail on some systems.

File: src/Behat/Behat/Gherkin/Specification/Locator/FilesystemFeatureLocator.php [hunk 4]
```diff
@@ -132,11 +132,17 @@ private function findFeatureFiles($path)
...
+                new RecursiveDirectoryIterator(
+                    $absolutePath,
+                    RecursiveDirectoryIterator::FOLLOW_SYMLINKS |
+                    RecursiveDirectoryIterator::SKIP_DOTS
```
Comment (stof) [inline src/Behat/Behat/Gherkin/Specification/Locator/FilesystemFeatureLocator.php]: this should be on the previous line to be more readable. It is the same argument, not a different argument

**Comment by sroze** [comment]
@everzet this PR sounds good and should be merged before the 3.1 release :+1:

**Comment by sherakama** [comment]
Thanks!

**Pull request merged**